### DISCUSSION TITLE
Mail consumer: delete not considered mails

### DISCUSSION THREAD

**Discussion by simonsilvalauinger** [jonaswinkler/paperless-ng#567] [Feature Requests]
Hello @jonaswinkler,
I'm glad you continued and improve paperless. It's a great tool which exactly fits my needs.

I created a dedicated mail address for the mail consumer and forward all received mails from my various mail accounts to it. Unfortunately the consumer can not delete mails without attachments or mails which have attachments that are not considered by it. At some point the mail account is full and can not receive mails anymore.
Would it be possible to add a flag in the mail rules to delete unconsidered mails?

Thank you
Simon

**Comment by jonaswinkler** [comment]
This sounds really dangerous to me, because people also use this on their actual mail accounts.

I won't be able to add that as an option to mail rules specifically, because:
- If we do, that rule might delete unmatched mails that would otherwise get matched by other rules.
- Therefore, this really should be an option you configure for each mail server, not mail rule.
- It should also probably be active only, if there's at least one mail rule.

I'll also have to check with the libraries about how to do it, since as it is right now, I'm only downloading messages that match the filter rules, and don't even know what mails exists besides that.
